core/arch/arm/mm/core_mmu.c

```diff
@@ -467,41 +556,64 @@ static bool pbuf_is_nsec_ddr(paddr_t pbuf __unused, size_t len __unused)
 #ifdef CFG_SECURE_DATA_PATH
 static bool pbuf_is_sdp_mem(paddr_t pbuf, size_t len)
 {
+	paddr_t base;
+	size_t size;
+
+	if (dtb_get_sdp_region(&base, &size))
```

Comment:
We should only rely on the DTB to be available during boot.

Reply:
do you mean that we should only define SDP from the DTB ?

Reply:
No, rather that we should extract what we need from the DTB during boot and then use that extracted information later on when needed. In this case I expect that we need to record the SDP ranges somewhere. With scattered arrays we already have the information recorded.

Reply:
ok, done here 75beb4be419a76808c12db1b906ad720f554b931